cluster-install/install.sh

```diff
@@ -1005,27 +1025,31 @@ check_if_image_can_be_pulled_via_curl() {
     if [ -z "$image_repo" ]; then fatal "no image_repo given"; fi
     if [ -z "$image_tag" ]; then fatal "no image_tag given"; fi
 
-    local -a args=('-s' '-w' "\n%{http_code}")
+    local -a args=('-sS' '-w' "\n%{http_code}")
     if [ -n "$encoded_creds" ]; then
         args+=("-H" "Authorization: Basic $encoded_creds")
     fi
+    args+=("-H" "Accept: application/vnd.docker.distribution.manifest.v2+json")
 
-    local -r result="$(curl -X GET "${args[@]}" "https://$registry/v2/$image_repo/tags/list")"
+    local -r result="$(curl -X GET "${args[@]}" "https://$registry/v2/$image_repo/manifests/$image_tag" 2> "$__ERR_FILE")"
+    local -r curl_err="$(get_captured_err)"
     local -r line_count="$(echo "$result" | wc -l)"
-    _return_body="$(echo "$result" | head -n "${line_count-1}")"
+    _return_body="$(echo "$result" | head -n "$((line_count-1))")"
     _return_status="$(echo "$result" | tail -n 1)"
     _return_error=""
 
     if [ "$_return_status" == 200 ]; then
-        # Example response body:
-        # {
-        #   "name" : "my/repo/astra-connector",
-        #   "tags" : [ "032172b", "074dk52", "1fbc135", "24.02" ]
-        # }
-        if echo "$_return_body" | grep "tags" | grep -q -- "$image_tag"; then
-            return 0
+        return 0
+    elif [ "$_return_status" == 404 ]; then
+        _return_error="the image was not found"
+    elif [ "$_return_status" != 000 ]; then
+        _return_error="$(echo "$_return_body" | jq -r '.errors.[0].message' 2> /dev/null)"
```

Comment:
This pulls the error message from the body, following the standard JSON path that docker registries return in the body when an error occurs. Can't fully rely on it though, so if we end up empty handed for whatever reason, we just set it to `unknown error` (next few lines).